mne/datasets/sleep_physionet/tests/test_physionet.py

```diff
@@ -78,14 +85,25 @@ def test_run_update_temazepam_records(tmpdir):
 
 
 @requires_good_network
-def test_sleep_physionet_temazepam(physionet_tmpdir):
+def test_sleep_physionet_temazepam(physionet_tmpdir, mocker):
     """Test Sleep Physionet URL handling."""
+    mm = mocker.patch('mne.datasets.sleep_physionet._utils._fetch_file',
```

Comment:
I don't get how this can work. mocker is not imported. Your mocker function does not write any file to disk

Reply:
what do you mean that the mocker is not imported? `pytest-mock` exposes this fixture called `mocker` which takes care of setting it up and tearing down.

And I'm aware that I'm writing nothing. But `_fetch_one` delegates all the writing to the original `_ferch_file` and I'm bypassing this so `_fetch_one` works just as expected despite `_fake_fetch_fle` does not write a thing.